app/javascript/techradar.js

```diff
@@ -357,93 +422,121 @@
 
             function highlightSpot(spot) {
                 var animationSpeed = '200';
+                var spotExpandedSize = 18;
 
                 // fade out all spots ...
                 d3.selectAll("circle.spot")
                     .attr('fill-opacity', 0.4)
                     .attr("stroke-opacity", 0.4);
 
+                var selectedSpot = d3.select("circle#" + asId(spot.title));
                 // ... except for our selection-candidate
-                d3.select("circle#" + asId(spot.title))
-                    .attr('fill-opacity', 1)
+                selectedSpot.attr('fill-opacity', 1)
                     .attr("stroke-opacity", 1);
 
                 // ... make our selection-candidate even bigger and animate that
-                d3.select("circle#" + asId(spot.title))
-                    .transition().ease('cubic-out').duration(animationSpeed)
+                selectedSpot.transition().ease('cubic-out').duration(animationSpeed)
                     .attr("stroke-width", 3)
-                    .attr("r", 18);
+                    .attr("r", spotExpandedSize);
```

Comment:
Quite a bunch of code here - would be better to encapsulate that snippet into a dedicated, well-named function (i.e. like with `showHistorySpots`